File: clock/milon_antony/clock.c
#include<stdio.h>
#include<string.h>
#include<time.h>
#include<stdlib.h>


struct tm *ct;

void gtime()
{
	time_t t;	
	t=time(NULL);
	ct=localtime(&t);
}


void disp(int h,int m)
{
	int f=0,g,fl=0,flg=0;
	char sh[10],sm[20],d[5],a[10];
	
	if(h==11)
		flg=1;
	
	
		
	
		if(m<=32)
		{
			f=m%10;
			g=m/5;
			
			if(f>=3&&f<8)
			{
				if(f<5)
				{
					g=(g+1)*5;
					strcpy(a,"ALMOST");
				}
				else
				{
					g*=5;
					strcpy(a,"AROUND");
				}
			}
			else if(f==5)
			{
				g*=5;
				strcpy(a,"PRECISELY");
			}
			else if(f==0)
			{
				g*=5;
				strcpy(a,"");
			}
			else
			{
				if(f>0&&f<3)
				{
					g*=5;
					strcpy(a,"AFTER");
				}
				else
				{
					g=(g+1)*5;
					strcpy(a,"ALMOST");
				}
			}
					
			strcpy(d," PAST");
			if(g==0&&f<3)
				strcpy(d,"");
			switch(g)
			{
				case 0:strcpy(sm,"");break;
				case 5:strcpy(sm,"FIVE");break;
				case 10:strcpy(sm,"TEN");break;
				case 15:strcpy(sm,"QUARTER");break;
				case 20:strcpy(sm,"TWENTY");break;
				case 25:strcpy(sm,"TWENTY-FIVE");break;
				case 30:strcpy(sm,"HALF");break;
			}
			if (h>=12)
				f=1;
			else f=0;
			h=h%12;
		}
		else if(m>32)
		{
			f=m%10;
			g=m/5;
			
			if(f>=3&&f<8)
			{
				if(f<5)
				{
					g=(g+1)*5;g=60-g;
					strcpy(a,"NEARLY");
				}
				else
				{
					g*=5;g=60-g;
					strcpy(a,"ABOUT");
				}
			}
			else if(f==5)
			{
				g*=5;g=60-g;
				strcpy(a,"");
			}
			else if(f==0)
			{
				g*=5;g=60-g;
				strcpy(a,"PRECISELY");
			}
			else
			{
				if(f>0&&f<3)
				{
					g*=5;g=60-g;
					strcpy(a,"AROUND");
				}
				else
				{
					g=(g+1)*5;g=60-g;
					strcpy(a,"NEARLY");
				}
			}
					
			strcpy(d," TO");
			if(g==0&&f>=8)
				strcpy(d,"");
			if (h>=12)
				f=1;
			else f=0;
			h=(h+1)%12;
			if(h==0)
				fl=1;
			
		
		switch(g)
		{
			case 0:strcpy(sm,"");break;
			case 5:strcpy(sm,"FIVE");break;
			case 10:strcpy(sm,"TEN");break;
			case 15:strcpy(sm,"QUARTER");break;
			case 20:strcpy(sm,"TWENTY");break;
			case 25:strcpy(sm,"TWENTY-FIVE");break;
			case 30:strcpy(sm,"HALF");break;
		}
		
	
	}
	switch (h)
	{
		case 0:strcpy(sh,"TWELVE");break;
		case 1:strcpy(sh,"ONE");break;
		case 2:strcpy(sh,"TWO");break;
		
		case 3:strcpy(sh,"THREE");break;
		case 4:strcpy(sh,"FOUR");break;
		case 5:strcpy(sh,"FIVE");break;
		
		case 6:strcpy(sh,"SIX");break;
		case 7:strcpy(sh,"SEVEN");break;
		case 8:strcpy(sh,"EIGHT");break;
		
		case 9:strcpy(sh,"NINE");break;
		case 10:strcpy(sh,"TEN");break;
		case 11:strcpy(sh,"ELEVEN");break;
	}
	if (((f==0&&h==0)||fl==1)&&flg==0)
		strcpy(sh,"MIDNIGHT");
		
	if (system("cls")) system("clear");
	
	printf("IT'S %s\n",a);
	printf("%s%s\n",sm,d);
	printf("%s\n",sh);
}
	

int main()
{
	int h,m,s,a=1;
	
	do
	{
		m=s;
		gtime();
		h=ct->tm_hour;
		m=ct->tm_min;
		s=ct->tm_sec;		
		if(s==0||a==1)
			disp(h,m);
		a=0;
	}while(1);


	return 0;
}

File: clock/krishnaraj_sekhar_pv/clock.py
import time
import random

#Uses the same dictionaries and list defined in http://helvetictoc.com/css/main.css
HOURS = ['twelve','one','two','three','four','five','six','seven','eight','nine','ten','eleven']
MINUTES = {'5':  'five past','10': 'ten past','15': 'quarter past','20': 'twenty past','25': 'twenty-five past','30': 'half past','35': 'twenty-five to', '40': 'twenty to','45': 'quarter to','50': 'ten to','55': 'five to'}
PREPOSITIONS = {'-1': ['almost', 'nearly'],'0': ['exactly', 'precisely', 'now', ''],'1': ['after', 'about', 'around']}
roundAbout = 'Its around midnight'
SPECIAL_CASES = {'23:58': roundAbout,'23:59': roundAbout,'0:0': 'Its midnight','0:1': roundAbout,'0:2': roundAbout,'12:0': 'its noon.'}

#Get Fuzzy Factor. 
def getFuzzyFactor(m):
 	ff = m % 5
	if ff==1:
		return '1'
	elif ff==2:
		return '1'
	elif ff==3:
		return '-1'
	elif ff==4:
	        return '-1'
	else:
		return '0'

#Get prepositions given the time. Uses Fuzzy Factor for calculation
def getPrep(m):
	ff = getFuzzyFactor(m)
	p = PREPOSITIONS[ff]
	r =random.randrange(0, len(p)-1)
	return p[r]

#Get the rounded time
def getRM(m):
	ff = getFuzzyFactor(m)
	if ff=='1' or ff=='0':
		mf = (int(m/5) * 5) 
	else:
		mf = int(m/5) * 5 + 5
	return mf



time_now = time.localtime()
h=time_now.tm_hour
m=time_now.tm_min
stc = str(h) + ":" + str(m)
print stc


try:
	result = SPECIAL_CASES[stc]
	print "result" ,result
except KeyError:
	if h>=12:
		h=h-12
	p = getPrep(m)
	RM = str(getRM(m))
	mw = str(MINUTES[RM])
	if m<=32:
		hw = HOURS[h]
	else:
		if h==11:
			hw = HOURS[0]
		else:
			hw = HOURS[h+1]
	result = "Its " + p + " " + mw + " " + hw
	print result

	
	




